core/iwasm/interpreter/wasm_loader.c

```diff
@@ -3458,6 +3528,14 @@ wasm_loader_load(const uint8 *buf, uint32 size, char *error_buf,
         goto fail;
     }
 
+#if (WASM_ENABLE_MULTI_MODULE != 0) && (WASM_ENABLE_LIBC_WASI != 0)
+    /* do a check about WASI Application ABI */
+    if (!check_wasi_abi_compatibility(module, main_module, error_buf,
```

Comment:
Only check the compatibility when module is in wasi mode, or error might occur when it is non-wasi module (e.g. compile with `/opt/wasi-sdk/bin/clang -nostdlib`):
```C
if (module->is_wasi_mode
    && !check_wasi_abi_compatibility(module, main_module, error_buf,
```

Reply:
We shall treat `clang -nostdlib` equal to `WASM_ENABLE_LIBC_WASI=0`. I mean we don't accept the third kind of .wasm beyond command/reactor model when `WASM_ENABLE_LIBC_WASI=1

Reply:
`LIBC_WASI` and `LIBC_BUILTIN` can be enabled at the same time.